src/tools/schema2kotlin.ts

```diff
@@ -55,33 +55,82 @@ ${withCustomPackage(`import arcs.Particle;
 import arcs.Entity
 import arcs.StringEncoder
 import arcs.StringDecoder
-`)}`;
+`)}
+import kotlin.reflect.KProperty\n`;
   }
 
   getClassGenerator(node: SchemaNode): ClassGenerator {
     return new KotlinGenerator(node);
   }
+
+  fileFooter(): string {
+    console.log(`child`);
+    return `
+class TextDelegate {
```

Comment:
Can these be defined once in the framework code (i.e. in Entity.kt)?

Reply:
These definitely should go somewhere else. Otherwise they will be copied for every entity we generate.

Reply:
(It will then also be really easy to add tests for them too)

Reply:
+1

Reply:
Fixed!